17.md

```diff
@@ -0,0 +1,154 @@
+NIP-17
+======
+
+Private Direct Messages
+-----------------------
+
+`draft` `optional`
+
+This NIP defines an encrypted direct messaging scheme using [NIP-44](44.md) encryption and [NIP-59](59.md) seals and gift wraps. 
+
+## Direct Message Kind
+
+Kind `14` is a chat message. `p` tags identify one or more receivers of the message.
+
+```js
+{
+  "id": "<usual hash>",
+  "pubkey": "<sender-pubkey>",
+  "created_at": now(),
+  "kind": 14,
+  "tags": [ 
+    ["p", "<receiver-1-pubkey>", "<relay-url>"],
+    ["p", "<receiver-2-pubkey>", "<relay-url>"],
+    ["e", "<kind-14-id>", "<relay-url>", "reply"] // if this is a reply
+    ["subject", "<conversation-title>"],
+    ...
+  ],
+  "content": "<message-in-plain-text>",
+}
+```
+
+`.content` MUST be plain text. Fields `id` and `created_at` are required. 
+
+Tags that mention, quote and assemble threading structures MUST follow [NIP-10](10.md). 
+
+Kind `14`s MUST never be signed. If it is signed, the message might leak to relays and become **fully public**.
+
+## Chat Rooms
+
+The set of `pubkey` + `p` tags defines a chat room. If a new `p` tag is added or a current one is removed, a new room is created with clean message history. 
+
+Clients SHOULD render messages of the same room in a continuous thread. 
+
+An optional `subject` tag defines the current name/topic of the conversation. Any member can change the topic by simply submitting a new `subject` to an existing `pubkey` + `p`-tags room. There is no need to send `subject` in every message. The newest `subject` in the thread is the subject of the conversation. 
```

Comment:
I just wanted to call this out because I actually do not know if it is reasonable or not. This restricts one chat room per unique set of pubkeys. That means they can't have different "rooms" for different subjects. Everything else in this section mentioned before this paragraph seems fine to me, though.

Reply:
Yep. It's mostly to keep it simple. there is no way to consistently use subject to define rooms ( missing events and posts in the past can mess It up). So we thought about following what Slack does and make the group about the people, not the topic.